### DISCUSSION TITLE
Tab completion directories level

### DISCUSSION THREAD

**Discussion by davidcortesortuno** [fish-shell/fish-shell#10413] [Q&A]
Hi, when I tap `tab` to complete/find a file name, after a command (like `git add`), fish shows me the files in the current directory and also all the subdirectories. Is there a way to make fish to only show files in a specified directory, but not subdirectories or parent directories?

**Comment by krobelus** [comment]
steps to reproduce:

```shell
$ mkdir -p foo
$ touch foo/(seq 100)
$ git add <Tab>
foo/1   (Untracked file)  foo/19  (Untracked file)  foo/37  (Untracked file)  foo/55  (Untracked file)  foo/73  (Untracked file)  foo/91           (Untracked file)
foo/2   (Untracked file)  foo/20  (Untracked file)  foo/38  (Untracked file)  foo/56  (Untracked file)  foo/74  (Untracked file)  foo/92           (Untracked file)
foo/3   (Untracked file)  foo/21  (Untracked file)  foo/39  (Untracked file)  foo/57  (Untracked file)  foo/75  (Untracked file)  foo/93           (Untracked file)
foo/4   (Untracked file)  foo/22  (Untracked file)  foo/40  (Untracked file)  foo/58  (Untracked file)  foo/76  (Untracked file)  foo/94           (Untracked file)
...
```

I think we should try to fix this. Perhaps "if a directory has no Git-tracked files inside, only show the directory, not its children".
I believe this would match Magit. The way it's implemented is by parsing `git status --porcelain`; that one does not descend into untracked dirs.

**Reply by faho** [reply]
> Perhaps "if a directory has no Git-tracked files inside, only show the directory, not its children".

We specifically do `git status --untracked-files=all` [here](https://github.com/fish-shell/fish-shell/blob/ce92472af1ef23a603656f11c460c634255afa4a/share/completions/git.fish#L171), because otherwise we would have to go from the current to-be-completed token. (which is less of an issue with `commandline -x`)

That's the option that controls this, not `--porcelain`.
